Core.ApplicationServices/AdviceService.cs

```diff
@@ -216,5 +234,78 @@ private void AddRecipientByRole(Advice advice, AdviceUserRelation r, MailMessage
                     break;
             }
         }
+
+        public void Delete(Advice advice)
+        {
+            DeleteJobFromHangfire(advice);
+            AdviceRepository.DeleteByKeyWithReferencePreload(advice.Id);
+            AdviceRepository.Save();
+        }
+
+        private void DeleteJobFromHangfire(Advice advice)
+        {
+            DeletePostponedRecurringJobFromHangfire(advice.JobId);
+            RecurringJob.RemoveIfExists("Advice: " + advice.Id);
+        }
+
+        private void DeletePostponedRecurringJobFromHangfire(string jobIdText)
+        {
+            var monitor = JobStorage.Current.GetMonitoringApi();
+            var jobsScheduled = monitor.ScheduledJobs(0, int.MaxValue).
+                Where(x => x.Value.Job.Method.Name == "CreateDelayedRecurringJob");
+            foreach (var j in jobsScheduled)
+            {
+                var t = j.Value.Job.Args[1].ToString(); // Pick "Advice: nn"
+                if (t.Contains(jobIdText))
+                {
+                    BackgroundJob.Delete(j.Key);
+                }
+            }
+        }
+
+        public void BulkDeleteAdvice(IEnumerable<Advice> toBeDeleted)
+        {
+            using var transaction = TransactionManager.Begin(IsolationLevel.ReadCommitted);
+            foreach (var advice in toBeDeleted)
+            {
+                DeleteJobFromHangfire(advice);
```

Comment:
De to linjer her er de samme som linje 240 ig 241. De bør konsolideres i en private method som kaldes af både delete og BulkDeleteAdvice. Ellers har vi bare to steder at vedligeholde næste gang vi skal udvide delete